### DISCUSSION TITLE
Settings: QUOTA_BYTES_PER_ITEM error when saving presets that together exceed 8K

### DISCUSSION THREAD

**Discussion by kah-ell** [coddingtonbear/obsidian-web#202] [Bug Report]
### Describe the bug

It seems that if your presets get long enough, chrome.storage is throwing a "options.tsx:29 Uncaught (in promise) Error: QUOTA_BYTES_PER_ITEM quota exceeded" error, but only in the javascript console. It looks like "Save Changes" did work, but the success window is just missing. Took me a while to find that one out. ;)

It even looks like the save worked because the textarea with the template has the changes, but if you try to use them they fail. If you export your settings and reimport them you'll see that your changes are gone.

Love your webclipper btw, I'm hooked and it has become part of my daily workflow.

### To Reproduce

1. Go to extension options
2. Write some longer Templates, it seems 8K is the limit
3. Click "Save Changes" on any of them
4. Notice the lack of a "Success Options Saved" window

### Expected Behavior

No size limit on templates. Well, at least more than 8K. :)

### Operating System

Linux Kubuntu 20.04

### Browser

chromium, vivaldi

### Browser Version

evergreen

**Comment by coddingtonbear** [comment]
TIL that there's apparently an 8KB limit for each key stored in `chrome.storage.sync`, and since all of the templates are stored in a single key, you're hitting that limit.  

As far as I know, you're the only one I've heard of hitting that limit, unfortunately; so my only advice for the near or medium term is to have fewer templates.  I know that's not a satisfying answer; so if you were hoping to get a better resolution, you could definitely feel free to file a pull request that works around that limitation.

I'm not sure what approach might be easiest, but there are a few different paths I can imagine:

1. We could store each template in its own key?
2. We could break the template data across multiple keys somehow (maybe via something like https://stackoverflow.com/a/73558894).

If you do decide to try putting together a PR, we should probably think/talk a little more about what options might be best before getting too far along, though.
